symengine/basic_to_upoly.cpp

```diff
@@ -0,0 +1,363 @@
+#include <symengine/visitor.h>
+
+namespace SymEngine
+{
+
+// returns the integer if `x` is a positive integer else
+// returns `-1` if negative integer, `0` if not an integer
+inline int positive_integer(const RCP<const Basic> &x)
+{
+    if (is_a<const Integer>(*x)) {
+        int tmp = rcp_static_cast<const Integer>(x)->as_int();
+        if (tmp > 0)
+            return tmp;
+        else
+            return -1;
+    }
+    return 0;
+}
+
+class UIntPolyGenaratorVisitor : public BaseVisitor<UIntPolyGenaratorVisitor>
+{
+private:
+    RCP<const Basic> res;
+    bool inpow;
+
+public:
+    RCP<const Basic> apply(const Basic &b, bool inpow_)
+    {
+        inpow = inpow_;
```

Comment:
If the current processing is happening in the exponent of a `Pow` or not. So that `x/2` fails, but `2**(x/2)` passes.